modules/service/main.tf

```diff
@@ -1 +1,1318 @@
+    }
+  }
+
+  dynamic "placement_constraints" {
+    for_each = var.placement_constraints
+
+    content {
+      expression = try(placement_constraints.value.expression, null)
+      type       = placement_constraints.value.type
+    }
+  }
+
+  # Set by task set if deployment controller is external
+  platform_version    = local.is_fargate && !local.is_external_deployment ? var.platform_version : null
+  propagate_tags      = var.propagate_tags
+  scheduling_strategy = local.is_fargate ? "REPLICA" : var.scheduling_strategy
+
+  dynamic "service_connect_configuration" {
+    for_each = length(var.service_connect_configuration) > 0 ? [var.service_connect_configuration] : []
+
+    content {
+      enabled = try(service_connect_configuration.value.enabled, true)
+
+      dynamic "log_configuration" {
+        for_each = try([service_connect_configuration.value.log_configuration], [])
+
+        content {
+          log_driver = try(log_configuration.value.log_driver, null)
+          options    = try(log_configuration.value.options, null)
+
+          dynamic "secret_option" {
+            for_each = try(log_configuration.value.secret_option, [])
+
+            content {
+              name       = secret_option.value.name
+              value_from = secret_option.value.value_from
+            }
+          }
+        }
+      }
+
+      namespace = lookup(service_connect_configuration.value, "namespace", null)
+
+      dynamic "service" {
+        for_each = try([service_connect_configuration.value.service], [])
+
+        content {
+
+          dynamic "client_alias" {
+            for_each = try([service.value.client_alias], [])
+
+            content {
+              dns_name = try(client_alias.value.dns_name, null)
+              port     = client_alias.value.port
+            }
+          }
+
+          discovery_name        = try(service.value.discovery_name, null)
+          ingress_port_override = try(service.value.ingress_port_override, null)
+          port_name             = service.value.port_name
+        }
+      }
+    }
+  }
+
+  dynamic "service_registries" {
+    # Set by task set if deployment controller is external
+    for_each = length(var.service_registries) > 0 ? [{ for k, v in var.service_registries : k => v if !local.is_external_deployment }] : []
+
+    content {
+      container_name = try(service_registries.value.container_name, null)
+      container_port = try(service_registries.value.container_port, null)
+      port           = try(service_registries.value.port, null)
+      registry_arn   = service_registries.value.registry_arn
+    }
+  }
+
+  tags                  = var.tags
+  task_definition       = local.task_definition
+  triggers              = var.triggers
+  wait_for_steady_state = var.wait_for_steady_state
+
+  timeouts {
+    create = try(var.timeouts.create, null)
+    update = try(var.timeouts.update, null)
+    delete = try(var.timeouts.delete, null)
+  }
+
+  depends_on = [aws_iam_policy.service]
```

Comment:
Did some more testing this morning, I think this might need to be depending on the policy attachment, not the policy itself? I managed to get myself into the task draining race condition.

Reply:
good point - I think this is what we do elsewhere as well - updated to depend on policy attachment